src/makefiles/default_rules.mk

```diff
@@ -3,11 +3,19 @@ SHELL := /bin/bash
 
 ifeq ($(KALDI_FLAVOR), dynamic)
   ifeq ($(shell uname), Darwin)
-    ifdef LIBNAME
-      LIBFILE = lib$(LIBNAME).dylib
+    ifdef ANDROIDINC # cross-compiling enabled on host MacOS
+      ifdef LIBNAME
+        LIBFILE = lib$(LIBNAME).so
+      endif
+      LDFLAGS += -Wl,-rpath -Wl,$(KALDILIBDIR)
+      EXTRA_LDLIBS += $(foreach dep,$(ADDLIBS), $(dir $(dep))$(notdir $(basename $(dep))).a)
```

Comment:
I'm surprised this is .a and not .so

Reply:
It looks like all the "kaldi-*.a" libraries will be generated along the way by current default make rules.  Since all the sub folders under `src` has already specified the .a files as its `ADDLIBS`, I simply take advantage of this to make the minimum changes.

Reply:
I've verified the changes on MacOS 10.14.5.